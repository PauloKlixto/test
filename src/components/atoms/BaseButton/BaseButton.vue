<template>
  <component
    :is="tag"
    type="button"
    v-bind="$attrs"
    class="BaseButton"
    :disabled="disabled"
    @click="click"
  >
    <span v-if="!!label" class="BaseButton__Text">
      {{ label }}
    </span>
  </component>
</template>

<script>
export default {
  name: 'BaseButton',
  props: {
    tag: {
      type: String,
      default: 'button',
      validator: val => ['button', 'a'].includes(val)
    },
    label: {
      type: String,
      default: ''
    },
    disabled: {
      type: Boolean,
      default: false
    }
  },

  methods: {
    click(e) {
      this.$emit('click', e)
    }
  }
}
</script>

<style lang="scss" scoped>
.BaseButton {
  height: 51px;
  -webkit-appearance: none;
  display: flex;
  background: inherit;
  padding: 0 10px;
  cursor: pointer;
  align-items: center;
  position: relative;
  white-space: nowrap;
  background: $color-red;
  border-radius: 10px;

  &:hover {
    filter: brightness(90%) contrast(150%);
  }

  &__Text {
    text-transform: uppercase;
    font-size: 16px;
    letter-spacing: 0.001rem;
    color: $color-white;
    text-align: center;
    display: block;
    width: 100%;
  }

}
</style>
