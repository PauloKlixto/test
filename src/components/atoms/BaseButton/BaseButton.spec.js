/* eslint-disable no-undef */
import { shallowMount } from '@vue/test-utils'
import BaseButton from './BaseButton'

describe('BaseButton', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(BaseButton)
  })

  test('Componente existe', () => {
    expect(wrapper.exists()).toBe(true)
  })

  describe('Clique no componente', () => {
    test('Chamar a função click', () => {
      const stub = jest.fn()
      wrapper.setMethods({ click: stub })

      wrapper.find('.BaseButton').trigger('click')
      expect(stub).toBeCalled()
    })

    test('Emitar funcao click', () => {
      wrapper.vm.click()
      expect(wrapper.emitted('click')).toBeTruthy()
    })
  })

  test('Componente não alterado', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
