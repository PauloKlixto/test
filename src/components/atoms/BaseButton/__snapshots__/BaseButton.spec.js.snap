// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton Componente n√£o alterado 1`] = `
"<button type=\\"button\\" class=\\"BaseButton\\">
  <!----></button>"
`;
